Damcilva/disable mariner unattended update default

dmcilvaney opened the pull request:
**What type of PR is this?**

<!--
Add one of the following kinds:
/kind bug
/kind test
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
<!-- 
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  <!-- Common commit types:
        build: Build 🏭
        chore: Maintenance 🔧
        ci: Continuous Integration 💜
        docs: Documentation 📘
        feat: Features 🌈
        fix: Bug Fixes 🐞
        perf: Performance Improvements 🚀
        refactor: Code Refactoring 💎
        revert: Revert Change ◀️
        style: Code Style 🎶
        security: Security Fix 🛡️
        test: Testing 💚 -->
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:


**Release note**:
```
none
```

cameronmeissner committed:
fix swapfile creation in certain mariner cases where a resource disk isn't attached (#2549)

Co-authored-by: Cameron Meissner <cameissner@microsoft.com>

ganeshkumarashok committed:
Update containerd version to 1.6.15 (#2552)

* Update containerd version to 1.6.15

* change containerd patch version to 0

* change patch version:

* Switch to 1.6.12-1

* Revert back to 1.6.15-1

AbelHu committed:
chore: Update new Windows containerd package to v0.0.56 (v1.6.14) and ignore hash values for k8s 1.26 package (#2556)

anujmaheshwari1 committed:
Release notes for release 2023.01.10 (#2558)

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>

anujmaheshwari1 committed:
Bumping image version to 2023.01.10 (#2557)

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>
Co-authored-by: Cameron <cameronmeissner@gmail.com>

yizhang4321 committed:
Add binaries for 1.23.15, 1.24.9, 1.25.5 k8s version. (#2559)

wanqingfu committed:
update windows base image for 1B (#2560)

chengliangli0918 committed:
release notes of ubuntu arm64 vhd 2022.12.15 and 2022.12.19 (#2563)

Co-authored-by: Charlie Li <charlili@microsoft.com>

HDYA committed:
weekly image cve patching from Sec Pipelines (#2564)

* patch image cves

* remove unnecessary preload

* should not introduce new version

AbelHu committed:
fix: Remove vfpOutput.txt before generating new logs (#2566)

jatakiajanvi12 committed:
Update Container Insights Image for January release (#2565)

* Omsagent may 2022 release

* Removing unncessary file changes

* Updating the ciprod images for January release

Co-authored-by: Ganga Mahesh Siddem <gangams@microsoft.com>
Co-authored-by: Janvi Jatakia <jajataki@microsoft.com>

dmcilvaney committed:
feat: disable unattended update for Mariner by default

hbeberman commented on parts/linux/cloud-init/artifacts/cse_main.sh:
```diff
@@ -305,6 +305,20 @@ else
+        # At 6:00:00 UTC (1 hour random fuzz) download and install package updates.
+        # Disable timer persistence so dnf-automatic-install doesn't run immediately on first boot.
+        sed -i 's/Persistent=true/Persistent=false/' /usr/lib/systemd/system/dnf-automatic-install.timer
+        systemctlEnableAndStart dnf-automatic-install.timer || exit $ERR_SYSTEMCTL_START_FAIL
```
You'll need to unmask the dnf-automatic-install service

hbeberman commented on vhdbuilder/packer/install-dependencies.sh:
```diff
@@ -40,6 +40,25 @@ APT::Periodic::Download-Upgradeable-Packages "0";
+  systemctl mask dnf-automatic-install.service || exit 1
+  systemctl disable dnf-automatic-install.service || exit 1
+  systemctl disable dnf-automatic-install.timer || exit 1
+  systemctl stop dnf-automatic-install.timer || exit 1
```
Might want to mask the timer as well here.

hbeberman commented on parts/linux/cloud-init/artifacts/cse_main.sh:
```diff
@@ -305,6 +305,20 @@ else
+        systemctl stop dnf-automatic-notifyonly.timer
+        # At 6:00:00 UTC (1 hour random fuzz) download and install package updates.
+        # Disable timer persistence so dnf-automatic-install doesn't run immediately on first boot.
+        sed -i 's/Persistent=true/Persistent=false/' /usr/lib/systemd/system/dnf-automatic-install.timer
```
We should check if this is still necessary. Since the timer is disabled for image creation it shouldn't need to immediately fire on first boot.

AbelHu committed:
fix: Update AKS Windows CSE Script Package to v0.0.22 (#2568)

gracewehner committed:
Update components.json (#2567)

andyzhangx committed:
upgrade Azure Disk CSI driver to v1.26.1 (#2569)

wanqingfu committed:
doc: Add release notes for AKS Windows VHD images 2023.1B (#2561)

* update release notes

* fix file name

paulgmiller commented on parts/linux/cloud-init/artifacts/cse_main.sh:
```diff
@@ -305,6 +305,20 @@ else
         systemctl restart --no-block apt-daily.service
         aptmarkWALinuxAgent unhold &
+    elif [[ $OS == $MARINER_OS_NAME ]]; then
+        {{- if EnableUnattendedUpgrade }}
```
nitpick. move this else to outside of the os check?

paulgmiller replied:
ah never mind ace has that download service outside of his {{- if EnableUnattendedUpgrade }}

paulgmiller commented on parts/linux/cloud-init/artifacts/cse_main.sh:
```diff
@@ -305,6 +305,20 @@ else
+        # Enable the automatic install timer and the check-restart timer.
+        # Stop the notify only dnf timer since we've enabled the auto install one.
+        # systemctlDisableAndStop adds .service to the end which doesn't work on timers.
+        systemctl disable dnf-automatic-notifyonly.timer
```
curious how it was notifying before. just writing out a messag to some file?

dmcilvaney replied:
By default it prints to `/etc/motd`, alternatively it can also e-mail, or print to stdio

paulgmiller commented on parts/linux/cloud-init/artifacts/cse_main.sh:
```diff
@@ -305,6 +305,20 @@ else
+        # At 6:00:00 UTC (1 hour random fuzz) download and install package updates.
+        systemctl unmask dnf-automatic-install.service || exit $ERR_SYSTEMCTL_START_FAIL
+        systemctl unmask dnf-automatic-install.timer || exit $ERR_SYSTEMCTL_START_FAIL
+        systemctlEnableAndStart dnf-automatic-install.timer || exit $ERR_SYSTEMCTL_START_FAIL
```
asssume systemctlEnableAndStart is alias on mareiner for enable not sure why @Ace used restart instead of start.

dmcilvaney replied:
They are helper functions, part of https://github.com/dmcilvaney/AgentBaker/blob/82f8ea4937c2cf8ead85b53d109fa836988ae574/parts/linux/cloud-init/artifacts/cse_helpers.sh#L225

If I had to guess, running without `--no-block` was causing issues with the apt service taking a long time to start. Might add that if anything shows up in testing, but since we are just starting/stopping times I don't expect to need to run asynchronously here. Since the helpers don't have a way to pass custom parameters in, manually calling systemctl was needed. (same issue we have with our .timers ).

AbelHu committed:
fix: Update Windows unit tests (#2571)

Commit:
fix: unpin mariner v1 image from apparmor issues (#2575)

* Revert "fix: pin mariner v1 to 2022.11.12 due to app armor issue (#2483)"

This reverts commit baaf54f3b2342d5f41ae76ecd09b082487898063.

* fix: unit tests for unpinned version

* fix: formatting

chmill-zz committed:
clearer cni naming (#2579)

yizhang4321 committed:
Add v1.23.15-hotfix.20230114 binaries to VHD. (#2574)

* Add v1.23.15-hotfix.20230114 binaries to VHD.

* Fix a typo: 1.23.15-v1.23.15-hotfix.20230114 ==> 1.23.15-hotfix.20230114

UtheMan committed:
Add Edge Zone SIG definition (#2581)

* init branch

* Add Edge Zone SIG definition (#2555)

* Add Edge Zone SIG definition

* Bumping edge zone image version to 2023.01.10

* Try not using ASC for edge zones

* Add new edge zones version

Co-authored-by: Mikolaj Umanski <mumanski@microsoft.com>
Co-authored-by: Mikołaj Umański <mik.umanski@gmail.com>

* remove unneeded error + handler

Co-authored-by: Mikolaj Umanski <mumanski@microsoft.com>
Co-authored-by: yunrugemicrosoft <114022099+yunrugemicrosoft@users.noreply.github.com>

hbeberman committed:
Update Mariner Kata VHDs to use marketplace image (#2582)

chmill-zz committed:
adding func for cni vhd steps and readding some that got dropped (#2580)

* adding func for cni vhd steps and readding some that got dropped

* fixing func name

* changing components.json to see if the tests pass like i think they should

* url into the func. Slight difference between cnis

* bad copy paste

UtheMan committed:
update go version (#2589)

Co-authored-by: Mikolaj Umanski <mumanski@microsoft.com>

anujmaheshwari1 committed:
Bumping image version to 2023.01.19 (#2587)

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>
Co-authored-by: Mikołaj Umański <mik.umanski@gmail.com>

andyzhangx committed:
upgrade azure file csi driver to v1.25.1

anujmaheshwari1 committed:
Release notes for release 2023.01.19 (#2590)

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>
Co-authored-by: Mikołaj Umański <mik.umanski@gmail.com>

Commit:
refactor: private cluster host agent without go templates (#2586)

UtheMan committed:
feat: support disable SSH (#2592)

* feat: support disable SSH

* style: make ssh config as an enum

* use iota for ssh enum

* make generate + new test case

Co-authored-by: Tongyao Si <tosi@microsoft.com>
Co-authored-by: Mikolaj Umanski <mumanski@microsoft.com>

cameronmeissner committed:
add parameterization for custom SIG version ID to AB E2E (#2544)

* add parameterization for custom vm image ID

* refactor, add SIG_VERSION_ID validation

* typo

* handle default command line args for local testing

* remove SIG_VERSION_ID from matrices, only rely on defaulting logic instead

* always specify sig image version id

* refactor

* bring back -u

* fix comments

* remove placeholder SIG_VERSION_ID from matrix.json

* remove -u from bash flags

Co-authored-by: Cameron Meissner <cameissner@microsoft.com>

chengliangli0918 committed:
remove special code path for arm64 ubuntu 18.04 (#2596)

* remove the special code path for arm64 18.04

* remove the special code path for arm64 18.04

Co-authored-by: Charlie Li <charlili@microsoft.com>

SeanDougherty committed:
Add support for Mariner 2.0 Gen 1 (#2594)

* Add support for Mariner 2.0 Gen 1

* remove gen1 naming from image build pipeline yaml

* remove duplicate packer json

ganga1980 committed:
container insights january 2023 agent release (#2595)

Commit:
refactor: custom cloud logic without go templ

Commit:
refactor: container runtime config without go template

Commit:
refactor: more function definitions, container runtime logic without go template

Commit:
refactor: dhcpv6 without go template

Commit:
refactor: move swap file without go template

Commit:
refactor: gpu driver version without go template

Commit:
refactor: temp disk doesn't need go templates

Commit:
refactor: reconcile-private-hosts without go template

Commit:
remove: dead krustlet code

Commit:
refactor: dhcpv6 without go template

Commit:
refactor: mig mode without go template

Commit:
refactor: aks custom cloud initialization without go template

Commit:
refactor: setup custom search domains without go template

Commit:
refactor: disable ssh without go template (#2603)

Commit:
refactor: proxy ca without go template (#2604)

Commit:
refactor: custom ca trust without go template (#2605)

alexeldeib committed:
refactor: teleport enablement without go template (#2607)

alexeldeib committed:
refactor: krustlet without go templates (#2608)

alexeldeib committed:
refactor: fabricmanager without go template (#2610)

alexeldeib committed:
refactor: docker private auth without go template (#2611)

cameronmeissner committed:
mariner2 -> marinerv2gen1 to work around ADO ArtifactTool bug (#2609)

* mariner2 -> marinerv2gen1 to work around ADO ArtifactTool bug

* update key name

Co-authored-by: Cameron Meissner <cameissner@microsoft.com>

alexeldeib committed:
refactor: remove sgx code (#2612)

alexeldeib committed:
refactor: ipv6 dual stack without go template (#2614)

cameronmeissner committed:
feat: use msi identity for final sas token generation (#2583)

* use msi identity for generating final sas token when building linux vhds for now

* typo

* modify expiration dates for token

* remove quotes

* try storage container name = vhd for test

* inject output storage container name via pipeline variables

* remove plumbing of classic storage account conn string

* quotes around vars in logging

* test az account show

* use az account show to determine whether using msi

* slight refactor

* ensure classica sa connection string

Co-authored-by: Cameron Meissner <cameissner@microsoft.com>

alexeldeib committed:
refactor: container runtime install without go template (#2615)

tamilmani1989 committed:
Include CNS and Cilium logs as part of collection script (#2616)

* include azure cns and cilium logs

* lint fix

* including generated files

* Generated files

* Fixed conflicting files

Co-authored-by: tamilmani1989 <tamanoha@DESKTOP-MQ3OU1R.localdomain>
Co-authored-by: Cameron <cameronmeissner@gmail.com>

alexeldeib committed:
style: remove go templated comments (#2617)

alexeldeib committed:
refactor: outbound command without go template (#2606)

alexeldeib committed:
refactor: unattended upgrade without go template (#2619)

alexeldeib committed:
refactor: no dupe bridge without go template (#2620)

alexeldeib committed:
refactor: swap file without go template (#2621)

alexeldeib committed:
refactor: huge pages without go template (#2622)

alexeldeib committed:
refactor: cloud check for china without go template (#2623)

alexeldeib committed:
refactor: cse file paths without go templ (#2624)

alexeldeib committed:
refactor: docker private server and custom host without go template (#2625)

alexeldeib committed:
refactor: convert exit code comments to bash from go template (#2626)

alexeldeib committed:
refactor: remove various go template references in cse_install (#2627)

alexeldeib committed:
refactor: remove go templates from func definitions (#2628)

alexeldeib committed:
Revert "refactor: remove go templates from func definitions (#2628)" (#2631)

This reverts commit 26b368e6e512968f1d8ab6d9014a559c412b940e.

Commit:
refactor: always define transparent huge page

Commit:
refactor: always define configureEtcEnvironment

Commit:
refactor: always define swap file function

Commit:
refactor: transparent huge pages without go template

Commit:
refactor: custom cloud, kubelet config without go template

Commit:
fix: gpu node conditional in cse main without go template

Commit:
refactor: custom ca trust file check without go template

Commit:
refactor: remove a few unnecessary go templates in cloud init

Commit:
refactor: remove unnecessary go templates from cloud init

- health monitors are entirely in vhd
- kubelet health monitor is never used
- apt preferences is an empty file also in vhd
- mount propagation flags is docker vhd only, docker no longer suppored

Commit:
refactor: always apply ip forwarding in vhd

Commit:
refactor: remove containerd exec_start from cloud init

Commit:
refactor: imds wireserver block into cse/vhd

Commit:
refactor: add reconcile-private-hosts script to vhd

Commit:
refactor: add mig partition script to vhd

Commit:
refactor: mig strategy out of cloud init

cameronmeissner committed:
switch to 1es msi for abe2e (#2662)

* switch to 1es msi for az login step

* do it in all the places

Co-authored-by: Cameron Meissner <cameissner@microsoft.com>

Commit:
refactor: fix broken wait_for_file due to missing EOF in file

cameronmeissner committed:
add build metadata tags to packer VMs (#2652)

* add build metadata tags to packer vm

* add tag to checkin pipeline

* quotes

* change case

* change case

Co-authored-by: Cameron Meissner <cameissner@microsoft.com>

ganeshkumarashok committed:
Move linux sig image version to config and change image bump pipeline (#2570)

* Update SIG version logic

* errors import

* switch from error to panic - because of init change. And jq nit

* fix return

* make variable private

anujmaheshwari1 committed:
release notes for 2023.01.25 (#2669)

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>

Commit:
refactor: proxy CA into CSE

Commit:
refactor: motd from cloud init to cse

Commit:
refactor: remove many unnecessary go templates

- systemd units can be on disk without being enabled
- same for scripts
- some of these things can be done unconditionally with no change like docker exec_start

Commit:
refactor: add kubelet service to vhd; fix indentation of arm64 mariner2

corvus-callidus committed:
feature: add /etc/os-release content to VHD release log (#2667)

Commit:
refactor: http proxy config entirely out of cloud init

Commit:
refactor: move bind-mount drop in to cse

Commit:
refactor: move ctr drop-in, cgroup drop in to cse

Commit:
refactor(fix): missing file name in drop in

Commit:
refactor: cleanup unused monitor functions

Commit:
refactor(fix): teleport conditional with two sets of brackets

Commit:
refactor: move custom sysctls to cse

Commit:
refactor: move kubelet config, tls bootstrap dropin to cse

Commit:
refactor: align docker/containerd cloud init templates

Commit:
refactor: kubelet config file drop in to cse

Commit:
refactor: move sp key, kubelet client key/cert to cse

ganeshkumarashok committed:
Move Mariner, 2204 and Edge Zone SIG versions to config (#2678)

* Move Mariner, 2204 and Edge Zone SIG versions to config

* Update Edge Zone SIG image version

Commit:
refactor: move uu enablement into cse

Commit:
refactor: write custom ca certs inside CSE

Commit:
refactor: remove more containerd cloud init conditionals

Commit:
refactor: kubelet default env file into cse

dmcilvaney committed:
Address feedback

alexeldeib committed:
bump: vhd 2023.01.25 (#2668)

chmill-zz committed:
adding a log for the nslookup section (#2676)

Commit:
refactor: add all missing core files to vhd

Commit:
refactor: add all missing non-core files to vhd

Commit:
refactor(fix): spacing typo

Commit:
refactor: move cluster ca into cse

Commit:
refactor: fix embedded go template in cse_main

nreisch committed:
Add gk 3.11.0 (#2699)

alexeldeib committed:
refactor: move containerd config to cse (#2701)

junjiezhang1997 committed:
chore: use self-created 1es hosted pools to build windows vhd (#2588)

* modify 1es pool for windows

* fix bug

* fix bug for release pipeline

* use variable in only one place to make it easier for change

Commit:
refactor: always put nvidia-modprobe in vhd, disabled

yunrugemicrosoft committed:
Update edge zone sig version to 2023.01.25 (#2708)

chengliangli0918 committed:
fix content of /proc/sys/crypto/fips_enabled on FIPS VHDs (#2709)

Co-authored-by: Charlie Li <charlili@microsoft.com>

UtheMan committed:
add 2023.01.20 and 2023.01.26 vhd notes (#2714)

Co-authored-by: Mikolaj Umanski <mumanski@microsoft.com>

cameronmeissner committed:
keep meta and latest versions of linux-{modules,image,headers} (#2712)

* only remove linux-modules on  non-fips during cleanup

* just change the regex

* fix regex

* use same regex in inverse match

* add --purge to apt-get

---------

Co-authored-by: Cameron Meissner <cameissner@microsoft.com>

junjiezhang1997 committed:
chore: test changes in windows cse package automatically (#2704)

* test changes in windows cse package automatically

* remove unecessary code

* resolve comments

* remove sleep in retrying to upload cse package

chengliangli0918 committed:
add stricter fips content checks (#2713)

* stricter fips content checks

* stricter fips content checks

---------

Co-authored-by: Charlie Li <charlili@microsoft.com>

ganeshkumarashok committed:
Bump VHD to 2023.02.01 (#2718)

ganeshkumarashok committed:
Release notes for 2023.02.01 VHD (#2719)

gracewehner committed:
remove oldest image and add new release image (#2721)

HDYA committed:
weekly image cve patching from Sec Pipelines (#2717)

* patch image cves

* Update components.json

wanqingfu committed:
chore: cache windows azure cni v1.4.41 (#2723)

* cache 1.4.41

* add 1.4.35 back

anlandu committed:
Update policy addon/webhook version to prod_20230203.1 (#2724)

andyzhangx committed:
upgrade to azure disk csi driver v1.26.2 (#2726)

* upgrade to azure disk csi driver v1.26.2

* remove v1.26.1

alexeldeib committed:
fix: skip outbound command if empty (#2728)

UtheMan committed:
restart containerd during provisioning if custom ca certs are passed (#2731)

* restart containerd during provisioning if custom ca certs are passed

* Add additional comments

---------

Co-authored-by: Mikolaj Umanski <mumanski@microsoft.com>

anujmaheshwari1 committed:
bump vhd to 0206 (#2734)

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>

cameronmeissner committed:
cleanup ab e2e custom SIG version ID scenario impl (#2732)

Co-authored-by: Cameron Meissner <cameissner@microsoft.com>

cameronmeissner committed:
fix ab e2e always reporting NodeReady regardless of actual status (#2737)

* cleanup ab e2e custom SIG version ID scenario impl

* fix ab e2e node readiness check

* add node describe call in failure cases for more verbosity

* add node describe call in failure cases for more verbosity

---------

Co-authored-by: Cameron Meissner <cameissner@microsoft.com>

anujmaheshwari1 committed:
add release notes for 0206 (#2739)

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>

tyler-lloyd committed:
chore: cache cni-dropgz:v0.0.4 (#2729)

Safe to remove v0.0.3 since there was a hotfix
that removed its usage in AKS RP. Currenlty,
v0.0.2 is the only version being used so keeping
it around until we need to rotate another version
in.

yizhang4321 committed:
Add binaries for k8s 1.26.1 version. (#2740)

AbelHu committed:
fix: Remove all download steps in Windows debug scripts (#2738)

anujmaheshwari1 committed:
remove unused cilium images (#2744)

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>

anlandu committed:
Update (#2746)

anujmaheshwari1 committed:
remove unused calico images (#2745)

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>

anujmaheshwari1 committed:
bump vhd to 0209 (#2748)

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>

anujmaheshwari1 committed:
notes for 0209 vhd (#2749)

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>

cameronmeissner committed:
restore repoDigests within image-bom.json for VHD builds (#2742)

* cleanup ab e2e custom SIG version ID scenario impl

* add digests to image-bom.json

* simplify

* typos

* errors...

* just cat it out

* fix id grep

* remove cat

* comments

* simplify, can only use ctr

* fix formatting

* cleanup

* make docker and containerd image-bom's consistent

* refactor with functions and constants

* finally fixed the aggregation...

---------

Co-authored-by: Cameron Meissner <cameissner@microsoft.com>

junjiezhang1997 committed:
chore: support WS2022 and WS2022-gen2 in AgentBaker Windows E2E (#2741)

* support WS2022 and WS2022-gen2 in AgentBaker Windows E2E

* add exit code for invalid case

andyzhangx committed:
upgrade azure file csi driver to v1.26.0

wanqingfu committed:
feat: one Windows VHD build mode (#2482)

* one mode using backfill deletion

* typo

* default backfill del to true in prod

* remove space

alexeldeib committed:
fix: sp file var undefined (#2751)

andyzhangx committed:
Revert "upgrade azure file csi driver to v1.26.0" (#2753)

This reverts commit 146a1a383bf75ecbe6b6cbe25ac35b079b5fb99e.

anujmaheshwari1 committed:
add kata TL sig config + release TL images weekly (#2757)

* add kata tl sig config

* fix naming

* gofmt

* pin to older good image for kata tl

---------

Co-authored-by: anujmaheshwari1 <amaheshwari@microsoft.com>

AbelHu committed:
chore: Update Windows Azure CNI to v1.4.43 (#2759)

* chore: Update Windows Azure CNI to v1.4.43

* Remove old Windows Azure CNI in cache

alexeldeib committed:
fix: indentation for non-tls bootstrapping kubeconfig (#2760)

ganeshkumarashok committed:
Update containerd version to 1.6.17-1 (#2758)

cameronmeissner committed:
persist cleanup-vhd.sh on disk for invocation in subsequent builds/releases (#2756)

* remove invocation of cleanup-vhd.sh from packer builds

* introduce RUN_VHD_CLEANUP pipeline var, test builds will run the cleanup normally, prod builds will no longer run this step

* fix mariner templates

* fix packer_source.sh

* simplify steps

* fix mariner again...

* change target dir of cleanup-vhd.sh

* simplify by removing conditional

---------

Co-authored-by: Cameron Meissner <cameissner@microsoft.com>

junjiezhang1997 committed:
update windows 2023 2b base image (#2763)

AbelHu committed:
feat: Always do not reboot in Windows CSE (#2754)

UtheMan committed:
add retry for gpu install (#2762)

* add retry for gpu install

* add quotes for array expansion to avoid splitting

---------

Co-authored-by: Mikolaj Umanski <mumanski@microsoft.com>

junjiezhang1997 committed:
fix (#2766)

AbelHu committed:
feat: Validate generated sas token before publishing artifacts (#2768)

* feat: Validate generated sas token

* fix typo

dmcilvaney committed:
Generated content after rebase

dmcilvaney committed:
Fix spacing

HDYA committed:
weekly image cve patching from Sec Pipelines (#2771)

* patch image cves

* Update components.json
